### PR TITLE
mm/iob: add support of partial bytes clone 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

1. mm/iob: add support of nonblock iob_clone
2. mm/iob: add support of partial bytes clone
3. net/devif: replace redundant code to iob_clone_partial()

## Impact

N/A

## Testing

iperf tcp client test

### CODE REVIEW

File: mm/iob/iob_clone.c [hunk 1]
```diff
@@ -137,26 +225,37 @@ int iob_clone(FAR struct iob_s *iob1, FAR struct iob_s *iob2, bool throttled)
        * transferred?
        */
 
-      if (offset2 >= CONFIG_IOB_BUFSIZE && iob1 != NULL)
+      if (offset2 >= (CONFIG_IOB_BUFSIZE -
+                      iob2->io_offset) && iob1 != NULL)
```
Comment (pkarashchenko) [inline mm/iob/iob_clone.c:229]:
```suggestion
      if (offset2 >= (CONFIG_IOB_BUFSIZE - iob2->io_offset) &&
          iob1 != NULL)
```

Comment (anchao) [reply mm/iob/iob_clone.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

File: mm/iob/iob_clone.c [hunk 2]
```diff
@@ -137,26 +225,37 @@ int iob_clone(FAR struct iob_s *iob1, FAR struct iob_s *iob2, bool throttled)
...
+ * Description:
+ *   Duplicate (and pack) the data in iob1 in iob2.  iob2 must be empty.
+ *
+ ****************************************************************************/
+
+int iob_clone(FAR struct iob_s *iob1, FAR struct iob_s *iob2,
+              bool throttled, bool block)
+{
+  DEBUGASSERT(iob2->io_len == 0 && iob2->io_offset == 0 &&
+            iob2->io_pktlen == 0 && iob2->io_flink == NULL);
```
Comment (pkarashchenko) [inline mm/iob/iob_clone.c:257]:
```suggestion
  DEBUGASSERT(iob2->io_len == 0 && iob2->io_offset == 0 &&
              iob2->io_pktlen == 0 && iob2->io_flink == NULL);
```

Comment (anchao) [reply mm/iob/iob_clone.c:257]: Thanks ! Let me fix it on next patch.

**Pull request merged**